### ISSUE TITLE
Support for iMessages with multiple recipients

### ISSUE THREAD

**Issue by vicould** [toffer/iphone-sms-backup#2]
Around the line 401 in the function responsible for getting the addresses for the iMessage a check is failed if the madrid_flags is different from the tested values, making the return of the function faulty as the variables are not initialized.
The code for this type of iMessage is 102405, I will submit a patch once it works for me. I have to probably look into how the recipients are retrieved from the item, so that the multiple recipients are correctly handled.

**Comment by toffer** [comment]
Yeah, handling group messages is more complicated than messages with single recipients, so I haven't gotten around to that.

However, just because I don't export group messages, that doesn't mean that the script should crash when it encounters one.  Which it does, currently.

That's a big screwup on my part, but I'm going to push a fix for that shortly.

As for actually exporting iMessage group chats, the catch is that the recipients aren't stored in the `message` table (like they are for single recipients).  Instead, they are stored as binary-plist BLOBs in the `participants` field in the `madrid_chat` table.  

Exporting group messages is something I would like to add, but I probably won't get to it right away.  If you add it before I do, I'd love to see what you come up with!

**Comment by vicould** [comment]
Thanks for the news! I just saw your commit, time to update.

I will try to do that during the week-end, it's always nice to hack some python :)
You also put in the read me that you skip the messages containing something else than text, do you know how the pictures are stored in the db? Is it directly data fields that would need a dump or is it more complex?

I also had a first issue due to the encryption of the backup, but I saw that Apple did not really document how they perform it, so there does no seem to be an easy solution.

Great work for the rest of the script, it seems well written, and not to hard to read and how the parts interact!

Ludovic
